contracts/cw-core/src/contract.rs

```diff
@@ -179,9 +184,13 @@ pub fn execute_proposal_hook(
     sender: Addr,
     msgs: Vec<CosmosMsg<Empty>>,
 ) -> Result<Response, ContractError> {
-    // Check that the message has come from one of the proposal modules
-    if !PROPOSAL_MODULES.has(deps.storage, sender) {
-        return Err(ContractError::Unauthorized {});
+    let module = PROPOSAL_MODULES
+        .may_load(deps.storage, sender.clone())?
+        .ok_or(ContractError::Unauthorized {})?;
+
+    // Check that the message has come from an active module
+    if module.status != ProposalModuleStatus::Enabled {
+        return Err(ContractError::ModuleAlreadyDisabled { address: sender });
```

Comment:
Tiny nit: maybe this should have its own error message? Seems `ModuleDisabled` or something?